_test/general.test.php

```diff
@@ -58,4 +78,408 @@ public function test_plugin_conf() {
+    }
+
+    /**
+     * Test to ensure that XML outputs basic export data.
+     */
+    public function test_xml_elements_equal_to_dokuwiki_page_data() {
+        $pageId = 'test123:test123:test123';
+        $pageTitle = 'test123';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        // Set title manually because you cant save it with the saveWikiText function.
+        $pageMetaTitle = array('title' => $pageTitle);
+        p_set_metadata($pageId, $pageMetaTitle);
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 20));
+
+        $names = $xml->xpath('/findologic/items/item/names/name');
+        $name = $names[0];
+
+        $summaries = $xml->xpath('/findologic/items/item/summaries/summary');
+        $summary = $summaries[0];
+
+        $descriptions = $xml->xpath('/findologic/items/item/descriptions/description');
+        $description = $descriptions[0];
+
+        $ordernumbers = $xml->xpath('/findologic/items/item/allOrdernumbers/ordernumbers/ordernumber');
+        $ordernumber = $ordernumbers[0];
+
+        $urls = $xml->xpath('/findologic/items/item/urls/url');
+        $url = $urls[0];
+
+        $properties = $xml->xpath('/findologic/items/item/allProperties/properties/property');
+        $propertyKey = (string)$properties[0]->key[0];
+        $propertyValue = (string)$properties[0]->value[0];
+
+        $attributes = $xml->xpath('/findologic/items/item/allAttributes/attributes/attribute');
+        $attributeKey = (string)$attributes[0]->key[0];
+        $attributeValue = (string)$attributes[0]->values[0]->value[0];
+
+        $dateAddeds = $xml->xpath('/findologic/items/item/dateAddeds');
+        $dateAdded = (string)$dateAddeds[0]->dateAdded[0];
+
+        $expectedName = $pageTitle;
+        $expectedSummary = $expectedDescription = self::PAGE_CONTENT_PLACEHOLDER;
+        $expectedOrdernumber = $pageId;
+        $expectedUrl = 'http://wiki.example.com/./doku.php?id=' . $pageId;
+        $expectedPropertyKey = 'dummy';
+        $expectedPropertyValue = 'dummy';
+        $expectedAttributeKey = 'cat';
+        $expectedAttributeValue = 'test123_test123_test123';
+        // Format DateTime because the creation date is not known. It can vary.
+        $pageMetadata = p_get_metadata($pageId);
+        $pageCreated = new DateTime();
+        $pageCreated->setTimestamp($pageMetadata["date"]["created"]);
+        $expectedDateAdded = (string)$pageCreated->format(\DateTime::ATOM);
+
+        $this->assertEquals($expectedName, $name, 'Expected name in XML should match the pages first title "test123".');
+        $this->assertEquals($expectedSummary, $summary, 'Expected summary in XML should match the pages content "This page is for test purposes.".');
```

Comment:
No need to write the expected value (`"This page is for test purposes.".`) in the assertion message. PHPUnit will print it out for you if there is a mismatch.

Reply:
Will add.